src/composables/useTokenGenerator/useTokenGenerator.ts

```diff
@@ -0,0 +1,82 @@
+import { ref } from 'vue';
+
+/**
+ * Token generator composable
+ */
+export default function useTokenGenerator() {
+    /**
+     * Character set for the access token
+     */
+    const characterSet =
+        'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*_+-=?';
+
+    /**
+     * Minimum token length
+     */
+    const minTokenLenght = ref(8);
+
+    /**
+     * Rules for the access token
+     */
+    const rules = {
+        required: (value: string) =>
+            !!value || 'An access token is required to start a session',
+        min: (v: string) =>
+            v.length >= minTokenLenght.value ||
+            `Min ${minTokenLenght.value} characters`,
+        uppercase: (v: string) =>
+            /[A-Z]/.test(v) || 'Must include at least one uppercase letter',
+        lowercase: (v: string) =>
+            /[a-z]/.test(v) || 'Must include at least one lowercase letter',
+        special: (v: string) =>
+            /[\W_]/.test(v) || 'Must include at least one special character',
+        number: (v: string) =>
+            /\d/.test(v) || 'Must include at least one number',
+    };
+
+    /**
+     * Generate a random token using crypto module in Node.js or browser depending on the environment (Node.js or browser) - browser for application and Node.js for tests and GitHub Actions
```

Comment:
Nice catch / workaround but is there no similar module that works in both environments?

Reply:
I tried one but it didn't really worked

Reply:
Alright, fine by me